Release/src/http/client/http_client_asio.cpp

```diff
@@ -986,12 +991,31 @@ class asio_context : public request_context, public std::enable_shared_from_this
         }
         if(m_openssl_failed)
         {
-            return verify_cert_chain_platform_specific(verifyCtx, host);
-        }
+
+            if (!is_end_certificate_in_chain(verifyCtx))
+            {
+                // Continue until we get the end certificate.
+                return true;
+            }
+
+            auto chainFunc = [this](const std::shared_ptr<certificate_info>& cert_info) {
+                return m_http_client->client_config().invoke_certificate_chain_callback(cert_info);
+            };
+
+            return http::client::details::verify_cert_chain_platform_specific(verifyCtx, utility::conversions::to_utf8string(host), chainFunc);
+            }
 #endif
 
         boost::asio::ssl::rfc2818_verification rfc2818(host);
-        return rfc2818(preverified, verifyCtx);
+        if(!rfc2818(preverified, verifyCtx))
+        {
+            return false;
+        }
+
+        auto info = std::make_shared<certificate_info>(host, get_X509_cert_chain_encoded_data(verifyCtx));
```

Comment:
This does currently not compile on Linux because `get_X509_cert_chain_encoded_data()` is not defined (due to some `#ifdef`). @chogorma Please refer to [our PR](https://github.com/chogorma/cpprestsdk/pull/1) to your private repository.